roles/openshift_hosted/tasks/registry.yml

```diff
@@ -43,7 +43,7 @@
 
 - name: Update registry environment variables when pushing via dns
   set_fact:
-    openshift_hosted_registry_env_vars: "{{ openshift_hosted_registry_env_vars | combine({'OPENSHIFT_DEFAULT_REGISTRY':'docker-registry.default.svc:5000'}) }}"
+    openshift_hosted_registry_env_vars: "{{ openshift_hosted_registry_env_vars | combine({'REGISTRY_OPENSHIFT_SERVER_ADDR':'docker-registry.default.svc:5000'}) }}"
```

Comment:
@sdodson we need a way to remove the old value (OPENSHIFT_DEFAULT_REGISTRY) from the set of env vars because the old value takes precedence over the new value.  Is there any example of how to do that?

Reply:
This code would only be invoked during an initial install. If we need to migrate from the old variable to the new variable we'll need to add a specific task to migrate that.

@kwoodson do we have an easy method to remove an env_var from a DC?

Reply:
we don't need to migrate (we continue to support the old value).

So if this is only going to run on new installs, then i think this is fine.

Do we currently have any code that attempts to migrate registry config/env today?

Reply:
@bparees The dockerregistry configuration is backward compatible. Never needed to migrate something.

Reply:
The only thing we're doing with the registry today during upgrade is updating the image tag to correspond to the upgraded version which triggers a re-deploy. I think back in 3.0/3.1 we made slight tweaks during upgrade but that code is long since dead.

Reply:
@legionus there would be a need to migrate if we're setting the new value and the old value is already set, and the user sets the new value to something different.  Their change will not take effect.

Since the deprecated key always takes precedence, that means that installer must always maintain the deprecated key on existing installs.

Anyway it's moot since apparently we don't ever "update" the registry's configuration today.  But if at some point we did want to update the registry's config during the ansible playbook run, we'd have to ensure we carried the deprecated key forward(updated it to the new value) or deleted it entirely so it was not used anymore.